docs/admin/out-of-resource.md

```diff
@@ -0,0 +1,220 @@
+---
+---
+
+* TOC
+{:toc}
+
+The `kubelet` needs to preserve node stability when available compute resources are low.
+
+This is especially important when dealing with incompressible resources such as memory or disk.
+
+If either resource is exhausted, the node would become unstable.
+
+## Eviction Policy
+
+The `kubelet` can pro-actively monitor for and prevent against total starvation of a compute resource.  In
+cases where it could appear to occur, the `kubelet` can pro-actively fail one or more pods in order to reclaim
+the starved resource.
+
+### Eviction Signals
+
+The `kubelet` can support the ability to trigger eviction decisions on the following signals:
+
+| Eviction Signal  | Description                                                                     |
+|------------------|---------------------------------------------------------------------------------|
+| memory.available | memory.available := node.status.capacity[memory] - node.stats.memory.workingSet |
+
+In future releases, the `kubelet` will support the ability to trigger eviction decisions based on disk pressure.
+
+Until that time, it is recommended users take advantage of [garbage collection](/docs/admin/garbage-collection/).
+
+### Eviction Thresholds
+
+The kubelet supports the ability to specify eviction thresholds that trigger the kubelet to reclaim resources.
+
+Each threshold is of the following form:
+
+`<eviction-signal><operator><quantity>`
+
+* valid `eviction-signal` tokens as defined above.
+* valid `operator` tokens are `<`
+* valid `quantity` tokens must match the quantity representation used by Kubernetes
+
+#### Soft Eviction Thresholds
+
+A soft eviction threshold pairs an eviction threshold with a required
+administrator specified grace period.  No action is taken by the `kubelet`
+to reclaim resources associated with the eviction signal until that grace
+period has been exceeded.  If no grace period is provided, the `kubelet` will
+error on startup.
+
+In addition, if a soft eviction threshold has been met, an operator can
+specify a maximum allowed pod termination grace period to use when evicting
+pods from the node.  If specified, the `kubelet` will use the lesser value among
+the `pod.Spec.TerminationGracePeriodSeconds` and the max allowed grace period.
+If not specified, the `kubelet` will kill pods immediately with no graceful
+termination.
+
+To configure soft eviction thresholds, the following flags are be supported:
+
+```
+--eviction-soft="": A set of eviction thresholds (e.g. memory.available<1.5Gi) that if met over a corresponding grace period would trigger a pod eviction.
```

Comment:
Why is this whole section in a code block? How about you remove the triple backticks and just have the flag names be surrounded in `backticks` so that a proper list renders here.